otherlibs/systhreads/st_win32.h

```diff
@@ -24,7 +24,242 @@ Caml_inline void st_msleep(int msec)
   Sleep(msec);
 }
 
-#include "st_pthreads.h"
+/* POSIX thread implementation of the "st" interface */
+
+#include <errno.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <signal.h>
+#include <time.h>
+#include <caml/osdeps.h>
+
+
+typedef HANDLE st_thread_id;
+
+/* Thread creation. Created in detached mode if [res] is NULL. */
+static int st_thread_create(st_thread_id * res,
+                            unsigned ( WINAPI *start_address )( void * ),
+                            void * arg)
+{
+  st_thread_id thr;
+  thr = (st_thread_id) _beginthreadex(
+    NULL, /* security: handle can't be inherited */
+    0,    /* stack size */
+    start_address,
+    arg,
+    0,    /* run immediately */
+    NULL  /* thread identifier */
+    );
+  if (thr == 0) {
+    return errno;
+  } else if (res == NULL) {
```

Comment:
I think chaining the tests like this is ugly. I would prefer the test for failure to be separated from the test to decide whether to detach the thread or not, as these are completely unrelated operations. In other words:
```suggestion
  }
  
  if (res == NULL) {
```

Reply:
Makes sense, thanks!